### PR TITLE
allow withdrawal of non-fee link

### PR DESCRIPTION
**Pull request by RensR**
## Solution

### CODE REVIEW

File: contracts/src/v0.8/ccip/onRamp/EVM2EVMOnRamp.sol [hunk 1]
```diff
@@ -826,17 +826,24 @@ contract EVM2EVMOnRamp is IEVM2AnyOnRamp, ILinkAvailable, AggregateRateLimiter,
...
   /// @param feeToken The token to withdraw
   /// @param to The address to send the tokens to
   function withdrawNonLinkFees(address feeToken, address to) external onlyOwnerOrAdmin {
-    if (feeToken == i_linkToken || to == address(0)) revert InvalidWithdrawParams();
+    if (to == address(0)) revert InvalidWithdrawParams();
 
-    // We require the link balance to be settled before allowing withdrawal
-    // of non-link fees.
-    if (_linkLeftAfterNopFees() < 0) revert LinkBalanceNotSettled();
+    // We require the link balance to be settled before allowing withdrawal of non-link fees.
```
Comment (matYang) [inline contracts/src/v0.8/ccip/onRamp/EVM2EVMOnRamp.sol:836]: nit: let's update the comment and say `We require sufficient link balance to pay nops`

File: contracts/src/v0.8/ccip/test/onRamp/EVM2EVMOnRamp.t.sol [hunk 2]
```diff
@@ -1525,6 +1525,32 @@ contract EVM2EVMOnRamp_withdrawNonLinkFees is EVM2EVMOnRampSetup {
     s_onRamp.withdrawNonLinkFees(address(s_token), address(this));
   }
 
+  function testWithdrawalOnlyLeftoverLinkSuccess() public {
```
Comment (matYang) [inline contracts/src/v0.8/ccip/test/onRamp/EVM2EVMOnRamp.t.sol]: it's a good candidate for a fuzz test

File: core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_v1_2_0.go [hunk 3]
```diff
@@ -37,85 +35,20 @@ const (
 )
 
 func init() {
-	onRampABI, err := abi.JSON(strings.NewReader(evm_2_evm_onramp.EVM2EVMOnRampABI))
+	onRampABI, err := abi.JSON(strings.NewReader(evm_2_evm_onramp_1_2_0.EVM2EVMOnRampABI))
```
Comment (matYang) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_v1_2_0.go:38]: Would it be a good idea to add like a `legacy, do not edit` comment in these old readers?
We should move these readers into a legacy folder sometime

Comment (RensR) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/onramp_v1_2_0.go:38]: I proposed to move all readers into folders based on version. Kinda the same idea, can do in a follow up as it's an unrelated big refactor

**Review by matYang — APPROVED** [review]

**Pull request merged**